enumerables.rb

```diff
@@ -0,0 +1,199 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength, Style/RedundantSelf, Style/GuardClause, Style/IfUnlessModifier, Metrics/BlockNesting, Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    array = self if self.class == Array
+    array = self.to_a if self.class == Range
+    i = 0
+    while i < array.length
+      if block_given?
+        yield array[i]
+      else
+        return "\#<Enumerator: #{self}:my_each>"
+      end
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    array = self if self.class == Array
+    array = self.to_a if self.class == Range
+    value = 0
+    while value < array.length
+      if block_given?
+        yield(array[value], value)
+      else
+        return "\#<Enumerator: #{self}:my_each_with_index>"
+      end
+      value += 1
+    end
+    array
+  end
+
+  def my_select
+    array = []
+    i = 0
+    while i < self.length
+      if yield self[i]
+        array.push(self[i])
+      end
+      i += 1
+    end
+    array
+  end
+
+  def my_all?(*args)
+    if args.length.zero?
+      i = 0
+      while i < self.length
+        if block_given?
+          if yield(self[i]) == false
+            return false
+          end
+        elsif self[i].nil?
+          return false
+        else
+          return true
+        end
+        i += 1
+      end
+    else
+      return false
+    end
+    true
+  end
+
+  def my_any?(*args)
+    if args.length.zero?
+      i = 0
+      while i < self.length
+        if block_given?
+          if yield self[i]
+            return true
+          end
+        else
+          return true
+        end
+        i += 1
+      end
+    else
+      j = 0
+      while j < self.length
+        if args[0] == self[j]
+          return true
+        end
+
+        j += 1
+      end
+
+    end
+    false
+  end
```

Comment:
- [ ] my_any? returns false if the block ever returns a value false or nil in a range,

- [ ] when no block or argument is given returns false if at least one of the collection is not true

- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class:
ex: (array.my_any?(Numeric)) should be equal array.any?(Numeric)

- [ ] when a Regex is passed as an argument returns true if any of the collection matches the Regex:
ex: (words.my_any?(/d/))should be equal words.any?(/d/)